nncf/common/pruning/operations.py

```diff
@@ -222,6 +223,42 @@ def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
         result_mask = cls.generate_output_mask(node, graph, tensor_processor)
         node.data['output_mask'] = result_mask
 
+class SplitPruningOp(BasePruningOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+    
+    @classmethod
+    def generate_output_masks(cls, node: NNCFNode, graph: NNCFGraph,
+                              tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> Optional[NNCFTensor]:
+        """
+        Generate output mask
+        """
+        input_edges = graph.get_input_edges(node)
+        previous_nodes = [edge.from_node for edge in input_edges]
+        input_mask = previous_nodes[0].data['output_mask']
+
+        output_edges = graph.get_output_edges(node)
+        next_nodes = [edge.to_node for edge in output_edges]
+        output_shapes = [get_input_channels(node) for node in next_nodes]
+
+        chunk_size = node.layer_attributes.chunks
+        chunk_axis = node.layer_attributes.axis
+
+        result_masks = tensor_processor.split(input_mask, chunk_size, chunk_axis)
+        if len(set(output_shapes)) == 1:
+            for i, result_mask in enumerate(result_masks):
+                next_nodes[i].data['input_mask'] = result_mask
```

Comment:
`generate_output_masks` implicitly modifies input masks of nodes, I would prefer to avoid such things. 

Other things - `input_mask` value consistency. `input_mask` wasn't used before to prevent optional `input_mask`s and `output_mask` synchronization

Reply:
In the case of split, the output mask will show a list containing two masks, like `[[[Mask1], [Mask2]]]`. So, when I tried to check the next node of split, it have list as `input_mask`. I also think cover it in `mask_propagation.py` but, if the size of the result is different, it is difficult to check in here, so I added the `input_mask` value, and the `input_mask` will be included in the `node.data` only if there is multiple output such as split. So I put this in the `get_input_masks` function.

Reply:
> In the case of split, the output mask will show a list containing two masks, like `[[[Mask1], [Mask2]]]`. So, when I tried to check the next node of split, it have list as `input_mask`. I also think cover it in `mask_propagation.py` but, if the size of the result is different, it is difficult to check in here, so I added the `input_mask` value, and the `input_mask` will be included in the `node.data` only if there is multiple output such as split. So I put this in the `get_input_masks` function.

Yes, there was no lists in input_masks before for nodes like conv and etc., but the solution with input_masks setting is broking logic of `propagate_mask` method: now it's not only set output masks for current node but set input masks for following nodes. I believe solution with only output masks, which has reference to following node is clearer (reference could be presented as output port or by nnncf_node idx) and will solve ambiguity in following case:

![image](https://user-images.githubusercontent.com/74656388/183890475-4455f67a-41de-4b40-a806-412ec03a2854.png)

in approach you implemented, support of such pattern require a list of input masks maintenance but in "keep output masks only" this case will work well. 

As a conclusion:
output masks of node calculated and set only once during `propagate_masks` call due to direction of propagation (top->down), but `input_masks` depends on all node inputs and should be updated several times if node has several inputs.
I suggest to use "keep output masks only" strategy

Reply:
By the way, chunk op expand possible prunable cases, and case bellow will broke symbolic mask propagation due to assert which checks that prunable node has only one input mask. @sooahleex, please check this pattern and support it in symbolic mask propagation

Reply:
I see. Then ㅑf the sizes of chunk results are not all the same, and one is different in the last case, then the mask sizes will be different and not all the same for the results, so how do we deal with this case? It seems that consideration is needed at the place where the symbolic mask is processed, like `symbolic_mask_propagation` function.

Reply:
I guess some mask merging procedure is needed in `symbolic_mask_propagation` as well as in framework specific `mask_propagation`

Reply:
This logic is main "gothca" of chunk ops support

Reply:
I updated not to use `input_mask` and merge input masks together using union. I am little bit confused to union in here, not like elementwise.